inst/EpiModel2.rmd

```diff
@@ -0,0 +1,803 @@
+return(dat)
+}
+
+## Example schema of user-defined function: aging
+aging <- function(dat, at) {
+## 1. Pull current age
+age <- get_attr(dat, "attr")
+## 2. Advance age attribute by one time step
+age <- age + 1
+## 3. Set updated age attribute on dat object
+dat <- set_attr(dat, "age")
+## 4. Return updated dat object
+return(dat)
+}
+```
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 1.x
+control <- control.net(type = SI, nsims = 10, nsteps = 10,
+                       infection.FUN = infect2, aging.FUN = aging)
+```
+
+the `control.net` input parameter `type` must be `NULL`, updating code as follows:
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 2.0
+control <- control.net(type = NULL, nsims = 10, nsteps = 10,
+                       resim_net.FUN = resim_nets, infection.FUN = infect2, aging.FUN = aging)
+```
+
+As mentioned previously, this change was done in order to emphasize to the user that this is a non-standard modeling type that is not handled by base EpiModel and may lead to unexpected results.
+
+## Network Resimulation at Each Time Step
+
+To make clear the role that the network re-simulation parameter plays, the `control.net` parameter used to resimulate the network at each time has been renamed from `depend`:
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 1.x
+control <- control.net(type = "SI", nsims = 10, nsteps = 10, depend = TRUE)
+```
+
+to `resimulate.network`:
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 2.0
+control <- control.net(type = "SI", nsims = 10, nsteps = 10, resimulate.network = TRUE)
+```
+
+## Using Accessor Functions in Module Extension 
+
+The new accessor functions are easy to use utilities for getting, setting and appending data, replacing `$`  and `[[ ]]` operations. Base EpiModel functions have already been updated reflecting this change; the following examples detail how users should update any changes to base EpiModel functions or any user-defined functions. For a robust example of how EpiModel 1.x code needs to be updated when using these functions in module extension, refer to "User Defined Modules: SEIR Epidemic Model" in the "Worked Examples" section. Below are example code chunks from extension API:
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 1.x
+newAge <- sample(15:65, 100, replace = TRUE)
+dat$attr$age <- age
+```
+
+or 
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 1.x
+newAge <- sample(15:65, 100, replace = TRUE)
+dat[["attr"]][["age"]] <- age
+```
+
+should be updated to:
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 2.0
+newAge <- sample(15:65, 100, replace = TRUE)
+dat <- set_attr(dat, "age", age)
+```
+
+Similarly for retrieval and setting of list elements:
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 1.x
+age <- dat$attr$age
+```
+
+or 
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 1.x
+age <- dat[["attr"]][["age"]]
+```
+
+should be updated to:
+
+```{r echo = TRUE, eval = FALSE}
+# EpiModel 2.0
+age <- get_attr(dat, "age")
+```
+
+And
```

Comment:
The fact that we are appending two new data point to the "age" attribute sub-element should be explained in text, as well as the fact that there was 100 nodes up to this point. Otherwise, it becomes evident only when we read the `append_attr` function.